app/gui/src/project-view/components/visualizations/VisualizationHost.vue

```diff
@@ -6,13 +6,18 @@ import { provideVisualizationConfig } from '@/providers/visualizationConfig'
 import { Vec2 } from '@/util/data/vec2'
 import { ToValue } from '@/util/reactivity'
 
-const { visualization, data, size, nodeType } = defineProps<{
-  visualization?: string | object
-  data?: any
-  size: Vec2
-  nodeType?: string | undefined
-  overflow?: boolean
-  toolbarOverflow?: boolean
+// A single prop `params` is important to mitigate a bug in Vue that causes
+// inconsistent state when multiple props are present on the custom elements component.
+// TODO[ib]: Add a link to the issue.
+const props = defineProps<{
+  params: {
+    visualization?: string | object
+    data?: any
+    size: Vec2
+    nodeType?: string | undefined
+    overflow?: boolean
+    toolbarOverflow?: boolean
+  }
 }>()
```

Comment:
To simplify usages, we could destructure props (https://vuejs.org/guide/components/props#reactive-props-destructure) - I'm not sure if it works for a second level, but event `const { params } = defineProps<...` should help.